azerothcore/azerothcore-wotlk
# feat: reorder character selection screen

## Changes Proposed:
-  Add an `order` column to the `characters` table, allowing players to reorder their characters in the character selection screen

## Issues Addressed:

None

## SOURCE:


## Tests Performed:

- Applied SQL and C++ patches
- Tested on Windows 10 client and server


## How to Test the Changes:


1. Create multiple characters in an account, eg:
![image](https://user-images.githubusercontent.com/24357633/123288704-17c65900-d510-11eb-9492-2a1b22f51c11.png)
2. Change their `order` column, for example:
```sql
UPDATE `characters` SET `order` = 1 WHERE `name` = "One";
UPDATE `characters` SET `order` = 2 WHERE `name` = "Two";
UPDATE `characters` SET `order` = 3 WHERE `name` = "Three";
UPDATE `characters` SET `order` = 4 WHERE `name` = "Four";
```
3. Log out and log back in
4. The characters should be reordered according to the `order` column:
![image](https://user-images.githubusercontent.com/24357633/123289036-6247d580-d510-11eb-9b27-016113ce65b2.png)
5. (optional) If you want to reset back to the default order, use
```sql
UPDATE `characters` SET `order` = NULL WHERE `name` IN ("One", "Two", "Three", "Four");
```

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
